torchvision/transforms/v2/functional/_utils.py

```diff
@@ -23,15 +25,17 @@ def wrapper(inpt, *args, **kwargs):
     return wrapper
 
 
-def _register_kernel_internal(dispatcher, datapoint_cls, *, datapoint_wrapper=True):
+def _register_kernel_internal(dispatcher, input_type, *, datapoint_wrapper=True):
```

Comment:
Most of the diff in this whole file is just renaming parameters to reflect that we now support more than just datapoints. I'll highlight the parts were there is more.

Reply:
While we're at it, can we also find a way to raise a decent error when one does

```
@register_kernel(MyDatapoint, resize)
```

? I.e. reversing the order of the parameters. Not sure what it does in this PR but in `main` it doesn't complain although obviously it doesn't do the right thing. Would be nice to raise a proper error for that.

Reply:
Done in 1eaf82e7eaffd1d21bcd4f6b5b4a08b67452a119.